### PR TITLE
Parse refactoring

### PR DESCRIPTION
**Pull request by mturkia**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Trying to split the parse-xml.c into more sensible files. Sqlite database parsing is the main one to move away. The idea is to have only the XML parsing in this file and have parse.c to cover the common functions. And parse-db.c includes the sqlite DB parsing parts.

We are not quite there yet, but now is a good time to get some comments before I spend more time on this. Especially if this doesn't seem to clear things up...

### Additional information:

DM5 sample log and tests were done to notice possible breakage during the changes.
We are missing Cobalt test cases, but the 2 databases I have appear to be working. I would rather have proper test cases implemented, but cannot include other people's logs into our sources without permission...

### Mentions:

@dirkhh a quick sanity check would be appreciated

### CODE REVIEW

**Review by neolit123 — APPROVED** [review]
nice one!
lgtm

File: core/parse.c [hunk 1]
```diff
@@ -9,6 +9,9 @@
 #include "parse.h"
 #include "divelist.h"
 #include "device.h"
+#include "gettext.h"
```
Comment (dirkhh) [inline core/parse.c]: that seems unrelated :)

**Review by dirkhh — APPROVED** [review]
micro nit on one commit that's not even worth re-pushing the changes.
Looks great to me. Let's do this and see what happens :-)

**Comment by dirkhh** [comment]
Android build fails here:
```
In file included from /workspace/subsurface/core/parse.c:3:0:
/workspace/subsurface/core/membuffer.h:31:74: error: unknown type name 'va_list'
 extern __printf(2, 0) void put_vformat(struct membuffer *, const char *, va_list);

```

**Comment by mturkia** [comment]
Yep, I noticed that, but Android build doesn't work for me locally ATM. I wonder if stdarg.h would solve this one...

**Comment by torvalds** [comment]
Looks ok to me.

I do wonder if you should do the suunto/cobalt/shearwater/divinglog parsers as individual files too rather than put them all in "parse-db.c", but I don't really care deeply.

**Comment by dirkhh** [comment]
@mturkia - the Android build should be straight forward these days on a "typical Linux that is not Debian" system. I can try to debug this locally which might be faster than going through the CI (since that always builds all the dependencies, first).

@torvalds thanks for the review. I agree that in a next step it would be neat to separate those parsers even further, but I think this is an excellent first step in that direction.

**Comment by neolit123** [comment]
btw, would this need a `ReleaseNotes.txt` bump?

**Comment by dirkhh** [comment]
@mturkia include stdarg.h BEFORE membuffer.h - that fixes the build for me

**Comment by dirkhh** [comment]
@neolit123 there isn't any user visible change, is there? So what would go into the ReleaseNotes?

**Comment by dirkhh** [comment]
Android build is fixed, Mac build hasn't even started yet (but that passed before so there shouldn't be a problem). I will just go ahead and pull this now. :-)
